VWOSdk/Core/VWOClient.cs

```diff
@@ -676,6 +738,57 @@ public bool FlushEvents()
             LogInfoMessage.NoVariationAllocated(file, userId, campaignKey);
             return new UserAllocationInfo();
         }
+        private UserAllocationInfo normalizeAndFindWinningCampaign(List<BucketedCampaign> shortlistedCampaigns, BucketedCampaign calledCampaign,
+            UserStorageMap userStorageMap, Dictionary<string, dynamic> customVariables,
+           string campaignKey, string apiName, Dictionary<string, dynamic> variationTargetingVariables, bool initIntegration, string userId, string groupName, string groupId, bool disableLogs = false)
+        {
+            int currentAllocation = 0;
+            foreach (BucketedCampaign campaign in shortlistedCampaigns)
+            {
+                campaign.Weight = 100.0 / shortlistedCampaigns.Count;
+                int stepFactor = CampaignHelper.GetCampaignBucketingRange(campaign.Weight);
+                if (stepFactor != 0)
+                {
+                    campaign.StartRange = currentAllocation + 1;
+                    campaign.EndRange = currentAllocation + stepFactor;
+
+                    currentAllocation += stepFactor;
+                }
+            }
+            double bucketValue = CampaignAllocator.GetUserHashForCampaign(userId, Convert.ToInt32(groupId));
+            BucketedCampaign winnerCampaign = CampaignHelper.getAllocatedItem(shortlistedCampaigns, bucketValue);
+            if (winnerCampaign != null && winnerCampaign.Id.Equals(calledCampaign.Id))
+            {
+                LogInfoMessage.GotWinnerCampaign(file, campaignKey, userId, groupName.ToString(), disableLogs);
+                return evaluateTrafficAndGetVariation(winnerCampaign, userStorageMap, customVariables, campaignKey, apiName, variationTargetingVariables, initIntegration, userId);
+            }
+            else
+            {
+                LogInfoMessage.CalledCampaignNotWinner(file, campaignKey, userId, groupName.ToString(), disableLogs);
+                return new UserAllocationInfo();
+            }
+        }
+        private Dictionary<string, dynamic> getEligibleCampaigns(List<BucketedCampaign> campaignList, string userId, string apiName, Dictionary<string, dynamic> customVariables)

```

Comment:
again, these methods could be part of utility file as these are not exposed to customer. Adding private methods will make this file too long

Reply:
method comments missing